packages/common/src/sites/_internal/convertCatalogToLegacyFormat.ts

```diff
@@ -0,0 +1,36 @@
+import { getProp } from "../../objects";
+import { getScopeGroupPredicate } from "../../search/utils";
+import { IModel } from "../../types";
+import { cloneObject } from "../../util";
+
+/**
+ * Converts the migrated catalog of a site model back into a legacy catalog format.
+ * As the new catalog format is much more flexible than the legacy format, only supported
+ * fields configurations (i.e., group ids) are transferred.
+ *
+ * @param modelToUpdate modified site model with a migrated catalog
+ * @param currentModel currently persisted site model with a legacy catalog
+ * @returns site model with any catalog group changes reflected in a legacy format
+ */
+export function convertCatalogToLegacyFormat(
+  modelToUpdate: IModel,
+  currentModel: IModel
+): IModel {
+  const updatedModel = cloneObject(modelToUpdate);
+  const updatedItemScope = getProp(modelToUpdate, "data.catalog.scopes.item");
+  const updatedGroupPredicate = getScopeGroupPredicate(updatedItemScope);
+  if (updatedGroupPredicate) {
+    const updatedLegacyCatalog = cloneObject(currentModel.data.catalog);
+    // TODO: Do we need to worry about whether predicate.group is an array or not? This applies here and in opendata-ui
```

Comment:
fwiw - It could also be a `MatchOptions` ({any: [], all: [], not: []}) - but that's highly unlikely, and I'm not sure how we'd express that in the legacy format (other than `any` which is how it's treated at the api level)

Reply:
@dbouwman that's a good point! Unfortunately, a lot of the search branches developed by others (Viv, Josh, etc.) also don't take into account the `MatchOptions` interface as it isn't always feasibly translated. Once things slow down we should definitely take a high level overview of what we want to do long term with our search interfaces. I'll create a tech issue to review this.